src/Lykke.MatchingEngine.Connector/Services/TcpOrderSocketService.cs

```diff
@@ -11,40 +11,61 @@ public class TcpOrderSocketService : ITcpClientService, ISocketNotifyer
         private readonly TasksManager<long, TheResponseModel> _tasksManager;
         private readonly TasksManager<string, TheNewResponseModel> _newTasksManager;
         private readonly TasksManager<string, MarketOrderResponseModel> _marketOrdersTasksManager;
+        private readonly ISocketLog _logger;
+        private readonly bool _ignoreErrors;
 
         public TcpOrderSocketService(TasksManager<long, TheResponseModel> tasksManager,
             TasksManager<string, TheNewResponseModel> newTasksManager,
-            TasksManager<string, MarketOrderResponseModel> marketOrdersTasksManager)
+            TasksManager<string, MarketOrderResponseModel> marketOrdersTasksManager,
+            ISocketLog logger = null,
+            bool ignoreErrors = false)
         {
             _tasksManager = tasksManager;
             _newTasksManager = newTasksManager;
             _marketOrdersTasksManager = marketOrdersTasksManager;
+            _logger = logger;
```

Comment:
Is it possible to add some "Null" logger if the "logger" is null and delete ? from all other places where _logger is used

Reply:
Yes, but it can be unexpected behavior for this library consumers. We already creating default logger  when using extensions methods (BindMeClient / AddMeClient), but TcpMatchingEngineClient is a public class with nullable "socketLog" parameter. May be we should implement your idea in a separate refactoring task.